includes/class-wc-tracker.php

```diff
@@ -540,6 +541,49 @@ private static function get_admin_user_agents() {
 		return array_filter( (array) get_option( 'woocommerce_tracker_ua', array() ) );
 	}
 
+	/**
+	 * Get order counts by payment method
+	 *
+	 * @return array
+	 */
+	public static function get_order_counts_by_payment_method() {
+		global $wpdb;
+
+		$orders_by_payment_method = $wpdb->get_results(
+			"
+			SELECT
+			    order_gateway.payment_method AS payment_method,
+			    COUNT( id ) AS orders_count
+			FROM (
+				SELECT
+					orders.ID AS id,
+				    MAX(
+				        CASE
+				            WHEN meta_key = '_payment_method'
+				            THEN meta_value
+				        END
+				    ) AS payment_method
+				FROM {$wpdb->prefix}posts AS orders
+				LEFT JOIN {$wpdb->prefix}postmeta AS order_meta ON order_meta.post_id = orders.ID
+				WHERE
+					orders.post_type = 'shop_order' AND
+					orders.post_status in ( 'wc-completed', 'wc-processing', 'wc-refunded' )
+				GROUP BY orders.ID
+			) AS order_gateway
+			WHERE payment_method IS NOT NULL
+			GROUP BY payment_method
+			ORDER BY orders_count DESC
+		"
+		);
```

Comment:
I'm sorry if it's apparent, but is there a reason to group the inner query by order id and picking the `MAX` from the group? 

```
SELECT
			    order_gateway.payment_method AS payment_method,
			    COUNT( id ) AS orders_count
			FROM (
				SELECT
					orders.ID AS id,
				    order_meta.meta_value AS payment_method
				FROM wp_posts AS orders
				LEFT JOIN wp_postmeta AS order_meta ON order_meta.post_id = orders.ID
				WHERE
					orders.post_type = 'shop_order' AND
					orders.post_status in ( 'wc-completed', 'wc-processing', 'wc-refunded' )
                AND meta_key = '_payment_method'
				AND meta_value IS NOT NULL
			) AS order_gateway
			
			GROUP BY payment_method
			ORDER BY orders_count DESC
```

It seems to me the query above should return the same set of results and should be slightly less costly, especially since it can ignore all those NULL values that would be otherwise created for all meta values. 

Is it because we want to be absolutely positive each order is counted only once or some other reason?

Conceptually, I don't like that this is a direct SQL query bypassing our CRUD abstraction, but since other tracker data is fetched in the same way, it's probably okay to leave it as it is, as it's more performant this way. We'll need to clean it up early next year anyway.

Reply:
You are right. I was trying to get the `payment method title` as well and eventually dropped the idea, as the title can be changed for each site depending on the admin's fancy. But the structure of the query has remained. I shall make the appropriate changes.